packages/admin-frontend/components/EditBook/EditTabBar.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+/**
+ * Part of GDL gdl-frontend.
+ * Copyright (C) 2018 GDL
+ *
+ * See LICENSE
+ */
+import * as React from 'react';
+import Router from 'next/router';
+import { withRouter } from 'next/router';
+import AppBar from '@material-ui/core/AppBar/AppBar';
+import Tab from '@material-ui/core/Tab/Tab';
+import Tabs from '@material-ui/core/Tabs/Tabs';
+
+type Props = {
+  router: {
+    pathname: string,
+    query: {
+      id: string,
+      lang: string
+    }
+  }
+};
+
+type State = {};
+
+class RouteAwareTabBar extends React.Component<Props, State> {
+  handleChange = (event: Event, selectedTab: number) => {
+    const {
+      router: { query }
+    } = this.props;
+
+    const editChapterSelected = selectedTab === 1;
+    const pathname = `/admin/edit/${editChapterSelected ? 'chapter' : 'book'}`;
+
+    Router.push({ pathname, query: { id: query.id, lang: query.lang } });
+  };
+
+  selectedTab = () => {
+    const {
+      router: { pathname }
+    } = this.props;
+    if (pathname === '/admin/edit/book') return 0;
```

Comment:
not too happy about this being order sensitive. I suggest that we add explicit value props to the tab components instead, such as `<Tab value="chapter" />`, and then this doesn't have to return integer constants